tools/azsdk-cli/Azure.Sdk.Tools.Cli/Services/Languages/LanguageRepoServiceFactory.cs

```diff
@@ -0,0 +1,126 @@
+using System.IO;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Azure.Sdk.Tools.Cli.Helpers;
+
+namespace Azure.Sdk.Tools.Cli.Services;
+
+/// <summary>
+/// Factory service for creating language-specific repository services.
+/// Detects the language of a repository and returns the appropriate service implementation.
+/// </summary>
+public class LanguageRepoServiceFactory
+{
+    /// <summary>
+    /// Creates the appropriate language repository service based on the detected language.
+    /// </summary>
+    /// <param name="packagePath">Absolute path to the package directory</param>
+    /// <param name="gitHelper">Git helper instance for repository operations</param>
+    /// <param name="logger">Optional logger instance for diagnostics</param>
+    /// <returns>Language-specific repository service</returns>
+    public static ILanguageRepoService CreateService(string packagePath, IGitHelper gitHelper, ILogger? logger = null)
+    {
+        logger?.LogInformation($"Create service for package at: {packagePath}");
+        if (string.IsNullOrWhiteSpace(packagePath))
+        {
+            throw new ArgumentException("Package path cannot be null or empty", nameof(packagePath));
+        }
+
+        if (!Directory.Exists(packagePath))
+        {
+            throw new DirectoryNotFoundException($"Package path does not exist: {packagePath}");
+        }
+
+        // Discover the repository root from the project path
+        var repoRootPath = gitHelper.DiscoverRepoRoot(packagePath);
+        logger?.LogInformation($"Discovered repository root: {repoRootPath}");
+        
+        // Create language service using factory (detects language automatically)
+        logger?.LogInformation($"Creating language service for repository at: {repoRootPath}");
+        var detectedLanguage = DetectLanguage(repoRootPath, logger);
+
+        return detectedLanguage switch
+        {
+            "python" => new PythonLanguageRepoService(packagePath, gitHelper, logger),
```

Comment:
I think each language service should have its own `.Create()` method so that we don't have to inject dependencies like gitHelper in the factory. Otherwise the factory dependency list will be the superset of all service/helper dependencies in the language services.